icu4c/source/common/locid.cpp

```diff
@@ -881,6 +894,9 @@ AliasData::loadData(UErrorCode &status)
     UDate end = uprv_getRawUTCtime();
     printf("AliasData::loadData took total %f ms\n", end - start);
 #endif  // LOCALE_CANONICALIZATION_DEBUG
+    if (U_FAILURE(status)) {
```

Comment:
optional: Same thing here -- if build() is well-behaved, then you could remove this failure check and the following nullptr check.

Reply:
no, I cannot, if I skip this status check, then the error would be set to U_MEMORY_ALLOCATION_ERROR while it is point to a different error, set inside builder.build()

Reply:
What I mean is, after your latest commit ... you should be able to shorten this function here to just:
```
void U_CALLCONV
AliasData::loadData(UErrorCode &status)
{
#ifdef LOCALE_CANONICALIZATION_DEBUG
    UDate start = uprv_getRawUTCtime();
#endif  // LOCALE_CANONICALIZATION_DEBUG
    ucln_common_registerCleanup(UCLN_COMMON_LOCALE_ALIAS, cleanup);
    AliasDataBuilder builder;
    gSingleton = builder.build(status);
#ifdef LOCALE_CANONICALIZATION_DEBUG
    UDate end = uprv_getRawUTCtime();
    printf("AliasData::loadData took total %f ms\n", end - start);
#endif  // LOCALE_CANONICALIZATION_DEBUG
}
```